src/runner_manager.py

```diff
@@ -0,0 +1,391 @@
+            os_name: Name of operating system.
+            arch_name: Name of architecture.
+
+        Returns:
+            Information on the runner application.
+        """
+        runner_bins: RunnerApplicationList = []
+        if isinstance(self.path, GitHubRepo):
+            runner_bins = self._github.actions.list_runner_applications_for_repo(
+                owner=self.path.owner, repo=self.path.repo
+            )
+        if isinstance(self.path, GitHubOrg):
+            runner_bins = self._github.actions.list_runner_application_for_org(org=self.path.org)
+
+        logger.debug("Response of runner binary list: %s", runner_bins)
+
+        try:
+            return next(
+                bin for bin in runner_bins if bin.os == os_name and bin.architecture == arch_name
+            )
+        except StopIteration as err:
+            raise RunnerBinaryError(
+                f"Unable query GitHub runner binary information for {os_name} {arch_name}"
+            ) from err
+
+    @retry(tries=5, delay=30, logger=logger)
+    def update_runner_bin(self, binary: RunnerApplication) -> None:
+        """Download a runner file, replacing the current copy.
+
+        Remove the existing runner binary to prevent it from being used. This
+        is done to prevent security issues arising from outdated runner binary
+        containing security flaws. The newest version of runner binary should
+        always be used.
+
+        Args:
+            binary: Information on the runner binary to download.
+
+        TODO:
+            Convert to download of runner binary to streaming to file, rather than save in memory
+            then copy to file. Ignore if the file size is too small.
+        """
+        logger.info("Downloading runner binary from: %s", binary.download_url)
+
+        # Delete old version of runner binary.
+        if self.runner_bin_path is not None:
+            self.runner_bin_path.unlink(missing_ok=True)
+            self.runner_bin_path = None
+
+        # Download the new file
+        response = self.session.get(binary.download_url, stream=True)
+
+        logger.info(
+            "Download of runner binary from %s return status code: %i",
+            binary.download_url,
+            response.status_code,
+        )
+
+        if binary.sha256_checksum:
+            hash = hashlib.sha256()
+
+        with tempfile.NamedTemporaryFile(delete=False) as tmp_file:
+            for chunk in response.iter_content(chunk_size=4096, decode_unicode=False):
+                tmp_file.write(chunk)
+
+                if binary.sha256_checksum:
+                    hash.update(chunk)
+
+        logger.info("Finished download of runner binary.")
+
+        # Verify the checksum if checksum is present.
+        if binary.sha256_checksum:
+            if binary.sha256_checksum != hash.hexdigest():
+                logger.error(
+                    "Mismatch of excepted hash of runner binary (%s) and calculated hash (%s)",
```

Comment:
To avoid the repetition of `of` I propose (and there's a typo for expected)
`The expected hash of runner binary (%s) doesn't match the calculated hash (%s)`